CANTroller2/src/devices.h

```diff
@@ -982,6 +1070,13 @@ class Simulator {
             auto kv = _devices.find(option); // look for the component
             if (kv != _devices.end()) {
                 can_sim = std::get<0>(kv->second); // if an entry for the component already existed, preserve its simulatability status
+                if (can_sim) { // if simulability has already been enabled...
+                    if (option == _pot_overload) { // ...and the pot is supposed to overload this component...
+                        d.set_source(ControllerMode::POT); // ...then set the input source for the associated Device to read from the pot
+                    } else if (_enabled) { // ...and the pot isn't overloading this component, but the simulator is running...
+                        d.set_source(ControllerMode::TOUCH); // ...then set the input source for the associated Device to read from the touchscreen
+                    }
+                }
             }
             d.set_pot_input(_pot);
```

Comment:
Nit the naming here is confusing, but I'm not sure there's a better way to do it. I love how we're using composability here. After seeing set_source(ControllerMode::POT) then set_pot_input(_pot), I was like set_source and set_input seem similar - one is the conceptual source for something do we use touch screen or potentiometer? The other is kind of setting/activating/linking our potentiometer, because otherwise it's like "What potentiometer?". Maybe we call it connect_pot_input(_pot) or register_pot_input(_pot) or assign_pot_input(_pot)? Maybe i'm just being crazy

Reply:
I think we should go with attach_potentiometer(pot), seems pretty clear to me. and it should call set_source() internally, so if you attach a pot it just works.

Reply:
actually after looking at it again, i don't think we want to set_can_source in attach_pot. right we rely on a Device to set whether or not it can be pot overloaded; if we call set_can_source somewhere else then we need to maintain a separate list of overridable devices somewhere, which is a hassle and error-prone. Instead, we should have a check in Simulator::register_device that checks whether a Device can use pot input before attaching the pot.